app/objects/secondclass/c_host.py

```diff
@@ -14,6 +22,10 @@ class HostSchema(ma.Schema):
     software = ma.fields.List(ma.fields.Nested(ServiceSchema()))
     os = ma.fields.Nested(OSSchema())
     mac = ma.fields.String()
+    freebie_abilities = ma.fields.Dict(keys=ma.fields.String(), values=ma.fields.List())
```

Comment:
Whats the sub-list for here?

Reply:
In this case, the sub list enables a few possibilities such as storing `fact` representations or storing the new edges or attributes which result from exploiting the host.  Referencing the example above, exploiting a freebie ability may result in a new `ability` or `adversary` which can be used against a different host in the network.  As it is a freebie_ability, we do not need to store the success probability, just the new information we gain by exploiting the ability.  This is another case where a dictionary of dictionaries could be appropriate, the list structure was selected as the `value` in the nested dictionary may be non-standard, such as a tuple of (`Host.ip`, `Host.access`) to represent another host on the network becoming available after exploiting the freebie ability.

Reply:
Ah lets isolate that. This is not the logic for path finding. I have an idea for how modular path finding logic will work, and when we get to that we may add more fields/data structure then.